### PR TITLE
Add Lro\Rpc Test

### PR DESCRIPTION
**Pull request by mcgallan**
# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: test/CadlRanchProjects.Tests/lro/rpc.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+using AutoRest.TestServer.Tests.Infrastructure;
+using Azure;
+using NUnit.Framework;
+
+namespace CadlRanchProjects.Tests.lro
+{
+    public class LongRunningOperation : CadlRanchTestBase
+    {
+        [Test]
+        public Task LongRunningRpc_Succeed() => Test(async (host) =>
```
Comment (archerzz) [inline test/CadlRanchProjects.Tests/lro/rpc.cs:16]:
```suggestion
        public Task LongRunningRpc_Completed() => Test(async (host) =>
```

File: test/CadlRanchProjects.Tests/lro/rpc.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        {
+            var option = new GenerationOptions("text");
+            var response = await new RpcClient(host, null).LongRunningRpcAsync(WaitUntil.Completed, option);
+            Assert.AreEqual(200, response.GetRawResponse().Status);
+            Assert.AreEqual("text data", response.Value.Data);
+        });
+
+        [Test]
+        public Task LongRunningRpc_InProgress() => Test(async (host) =>
```
Comment (archerzz) [inline test/CadlRanchProjects.Tests/lro/rpc.cs:25]:
```suggestion
        public Task LongRunningRpc_Started() => Test(async (host) =>
```

**Review by archerzz — APPROVED** [review]
LGTM. Need some tuning on the names. Thanks.

File: test/CadlRanchProjects.Tests/lro/rpc.cs (hunk 1 above)
Comment (mcgallan) [reply test/CadlRanchProjects.Tests/lro/rpc.cs]: fixed.

File: test/CadlRanchProjects.Tests/lro/rpc.cs (hunk 2 above)
Comment (mcgallan) [reply test/CadlRanchProjects.Tests/lro/rpc.cs]: fixed.

**Pull request merged**